pkg/provider/azure_vmss_cache.go

```diff
@@ -294,11 +294,18 @@ func (ss *ScaleSet) DeleteCacheForNode(nodeName string) error {
 		return err
 	}
 
-	err = timedcache.Delete(cacheKey)
+	entry, exists, err := timedcache.Store.GetByKey(cacheKey)
 	if err != nil {
-		klog.Errorf("DeleteCacheForNode(%s) failed with error: %v", nodeName, err)
 		return err
 	}
+	if exists {
```

Comment:
I see master branch does not have such issue, so this is a specific fix on 1.23?
https://github.com/kubernetes-sigs/cloud-provider-azure/blob/631b3a4199f1301759c1ff74d1fdbbd49ea47632/pkg/provider/azure_vmss_cache.go#L273

Reply:
this needs cherry-pick to 1.23-1.25.

Reply:
1.23-1.25